### PR TITLE
Add Product List Page and Card Mock

### PR DESCRIPTION
**Pull request by varun13169**
Add Product List Page and Card Mock

image

### CODE REVIEW

File: src/components/Card/Card.jsx [hunk 1]
```diff
@@ -12,36 +12,58 @@ export default function Card(props) {
     (() => {
       console.log("No Seconday Action Defined");
```
Comment (zoef-yt) [inline src/components/Card/Card.jsx:13]: Fix:- Remove console.log()

File: src/components/Card/Card.jsx [hunk 2]
```diff
@@ -12,36 +12,58 @@ export default function Card(props) {
...
-      </div>
+      {isOnSale && (
+        <div className="dui-badge">
+          <div className="dui-util-txt-sm dui-util-fw-reg">SALE</div>
+        </div>
+      )}
       {/* <!-- Badge Component Ends -- with Text --> */}
 
       <div className="dui-card-prod-hzntl__img-container">
         <img className="dui-card-prod-hzntl__img" src={imgg} alt="" />
```
Comment (zoef-yt) [inline src/components/Card/Card.jsx:49]: Changes:- Add alt tag as it is empty here.

File: src/pages/productListingPage/productListingPage.jsx [hunk 3]
```diff
@@ -14,12 +32,8 @@ export default function ProductListingPage() {
...
-          {[1,2,3,4,5,6,7,8].map((itemDetails) => {
-            // console.log(a);
-            return (
-              <Card
-              />
-            );
+          {productAndFilterState.productsToShow.map((product) => {
+            return <Card key={product.id} itemDetails={product} />;
           })}
           {/* <div className="mobl-fltr-btn-helper"></div>
```
Comment (zoef-yt) [inline src/pages/productListingPage/productListingPage.jsx:38]: remove commented code.

**Review by zoef-yt — APPROVED** [review]
The code looks great. The code is understandable. Looks good to me. Good to merge. Good work.

File: src/contexts/productFilterContext/productFilterContext.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
+import { createContext, useReducer, useContext } from "react";
+
+const ProductFilterContext = createContext();
+
+const ProductFilterContextProvider = ({ children }) => {
+  const productFilterReducer = (productAndFilterState, action) => {
+    const actionType = action.type;
+    const ww = action.data.orgProducts;
```
Comment (humzahasan) [inline src/contexts/productFilterContext/productFilterContext.js:8]: a bit more descriptive variable name would help

File: src/contexts/productFilterContext/productFilterContext.js [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+              return beta * (Number(b.salePrice) - Number(a.salePrice));
+            }
+          ),
+        };
+
+      case "DEFAULT_ALL_FILTERS":
+        return productAndFilterState;
+
+      default:
+        console.log("");
```
Comment (humzahasan) [inline src/contexts/productFilterContext/productFilterContext.js:42]: For switch case without default, try adding this. 

```
/*eslint default-case: "error"*/
```

**Review by humzahasan — APPROVED** [review]
Code looks good to merge! Can have a look at the suggestion.

**Pull request merged**